packages/react-debug-tools/src/__tests__/ReactHooksInspectionIntegrationDOM-test.js

```diff
@@ -0,0 +1,185 @@
+    ReactDOMClient = require('react-dom/client');
+    act = require('internal-test-utils').act;
+    ReactDebugTools = require('react-debug-tools');
+  });
+
+  // @gate enableFormActions && enableAsyncActions
+  it('should support useFormStatus hook', async () => {
+    function FormStatus() {
+      const status = ReactDOM.useFormStatus();
+      React.useMemo(() => 'memo', []);
+      React.useMemo(() => 'not used', []);
+
+      return JSON.stringify(status);
+    }
+
+    const treeWithoutFiber = ReactDebugTools.inspectHooks(FormStatus);
+    expect(normalizeSourceLoc(treeWithoutFiber)).toEqual([
+      {
+        debugInfo: null,
+        hookSource: {
+          columnNumber: 0,
+          fileName: '**',
+          functionName: 'FormStatus',
+          lineNumber: 0,
+        },
+        id: null,
+        isStateEditable: false,
+        name: '.useFormStatus',
+        subHooks: [
+          {
+            debugInfo: null,
+            hookSource: {
+              columnNumber: 0,
+              fileName: '**',
+              functionName: 'Object.useFormStatus',
+              lineNumber: 0,
+            },
+            id: null,
+            isStateEditable: false,
+            name: 'HostTransitionStatus',
+            subHooks: [],
+            value: null,
+          },
+        ],
+        value: null,
+      },
+      {
+        debugInfo: null,
+        hookSource: {
+          columnNumber: 0,
+          fileName: '**',
+          functionName: 'FormStatus',
+          lineNumber: 0,
+        },
+        id: 0,
+        isStateEditable: false,
+        name: 'Memo',
+        subHooks: [],
+        value: 'memo',
+      },
+      {
+        debugInfo: null,
+        hookSource: {
+          columnNumber: 0,
+          fileName: '**',
+          functionName: 'FormStatus',
+          lineNumber: 0,
+        },
+        id: 1,
+        isStateEditable: false,
+        name: 'Memo',
+        subHooks: [],
+        value: 'not used',
+      },
+    ]);
+
+    const root = ReactDOMClient.createRoot(document.createElement('div'));
+
+    await act(() => {
+      root.render(
+        <form>
+          <FormStatus />
+        </form>,
+      );
+    });
+
+    // Implementation detail. Feel free to adjust the position of the Fiber in the tree.
+    const formStatusFiber = root._internalRoot.current.child.child;
+    const treeWithFiber = ReactDebugTools.inspectHooksOfFiber(formStatusFiber);
+    expect(normalizeSourceLoc(treeWithFiber)).toEqual([
+      {
+        debugInfo: null,
+        hookSource: {
+          columnNumber: 0,
+          fileName: '**',
+          functionName: 'FormStatus',
+          lineNumber: 0,
+        },
+        id: null,
+        isStateEditable: false,
+        name: '.useFormStatus',
+        subHooks: [
+          {
+            debugInfo: null,
+            hookSource: {
+              columnNumber: 0,
+              fileName: '**',
+              functionName: 'Object.useFormStatus',
```

Comment:
In the build tests in prod this comes out as `Object.useHostTransitionStatus [as useFormStatus]` with source map tripping up debug-tools: all subsequent hooks (even built-in ones) are consider sub hooks. If I remove the source map the sub hooks parsing is at least correct because then we end up with `Object.<anonymous>.exports.useFormStatus`.

Reply:
The `[as ...]` comes from Node:

```js
const Dispatcher = {
  useFormStatus: function ia() {
    console.log(new Error().stack);
  },
};
Dispatcher.useFormStatus();
```
prints 
```
Error
    at Object.ia [as useFormStatus] (**)
```

Reply:
Seems like a Jest issue. When I run 
```js
const React = require("react");
const ReactDOM = require("react-dom");
const ReactDebugTools = require("react-debug-tools");

function Component() {
  ReactDOM.useFormStatus();
  React.useMemo(() => {}, []);
  React.useMemo(() => {}, []);

  return null;
}

console.log(JSON.stringify(ReactDebugTools.inspectHooks(Component), null, 2));

```
via
```bash
$  NODE_ENV=production node --enable-source-maps test.js
[
  {
    "id": null,
    "isStateEditable": false,
    "name": "FormStatus",
    "value": null,
    "subHooks": [
      {
        "id": null,
        "isStateEditable": false,
        "name": "HostTransitionStatus",
        "value": null,
        "subHooks": [],
        "debugInfo": null,
        "hookSource": {
          "lineNumber": 139,
          "functionName": "useFormStatus",
          "fileName": "/Users/sebastian.silbermann/react-debug-tools-useFormStatus/node_modules/react-dom/cjs/react-dom.production.js",
          "columnNumber": 31
        }
      }
    ],
    "debugInfo": null,
    "hookSource": {
      "lineNumber": 6,
      "columnNumber": 12,
      "functionName": "Component",
      "fileName": "/Users/sebastian.silbermann/react-debug-tools-useFormStatus/test.js"
    }
  },
  {
    "id": 0,
    "isStateEditable": false,
    "name": "Memo",
    "subHooks": [],
    "debugInfo": null,
    "hookSource": {
      "lineNumber": 7,
      "functionName": "Component",
      "fileName": "/Users/sebastian.silbermann/react-debug-tools-useFormStatus/test.js",
      "columnNumber": 9
    }
  },
  {
    "id": 1,
    "isStateEditable": false,
    "name": "Memo",
    "subHooks": [],
    "debugInfo": null,
    "hookSource": {
      "lineNumber": 8,
      "functionName": "Component",
      "fileName": "/Users/sebastian.silbermann/react-debug-tools-useFormStatus/test.js",
      "columnNumber": 9
    }
  }
]
```

I get the correct output.

Reply:
Doesn't work in a browser either though: 
![Screenshot 2024-04-03 at 20 18 57](https://github.com/facebook/react/assets/12292047/afdc3163-2f30-415a-9b64-1a787ebe1074)

-- https://react-devtools-hook-inspection-experimental.vercel.app/